buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -912,123 +912,98 @@ class BuildPlugin implements Plugin<Project> {
         }
     }
 
-    static void applyCommonTestConfig(Project project) {
-        project.tasks.withType(RandomizedTestingTask) {task ->
-            jvm "${project.runtimeJavaHome}/bin/java"
-            parallelism System.getProperty('tests.jvms', project.rootProject.ext.defaultParallel)
-            ifNoTests 'fail'
-            onNonEmptyWorkDirectory 'wipe'
-            leaveTemporary true
-            project.sourceSets.matching { it.name == "test" }.all { test ->
-                task.testClassesDirs = test.output.classesDirs
-                task.classpath = test.runtimeClasspath
-            }
-            group =  JavaBasePlugin.VERIFICATION_GROUP
-            dependsOn 'testClasses'
-
-            // Make sure all test tasks are configured properly
-            if (name != "test") {
-                project.tasks.matching { it.name == "test"}.all { testTask ->
-                    task.shouldRunAfter testTask
-                }
-            }
-            if (name == "unitTest") {
-                include("**/*Tests.class")
+    static void applyTestConfig(Project project) {
+        // Default test task should run only unit tests
+        project.pluginManager.withPlugin('java') {
+            project.tasks.named('test') {
+                include '**/*Tests.class'
             }
+        }
 
-            // TODO: why are we not passing maxmemory to junit4?
-            jvmArg '-Xmx' + System.getProperty('tests.heap.size', '512m')
-            jvmArg '-Xms' + System.getProperty('tests.heap.size', '512m')
-            jvmArg '-XX:+HeapDumpOnOutOfMemoryError'
+        // none of this stuff is applicable to the `:buildSrc` project tests
+        if (project.path != ':build-tools') {
```

Comment:
The logic lives in the plugin, so we'd have to expose some flag or DSL for project to tell the plugin "don't apply the test stuff". This might be a little overkill since this is an exception case. I do agree this is a bit coupled but it's not the only [instance](https://github.com/elastic/elasticsearch/blob/d968dd212346479ead31f8f8ceb085b753087ad1/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy#L517) of this [kind of thing](https://github.com/elastic/elasticsearch/blob/37768b7eac24ad226dc0e4b0d34614d98b33fcd2/buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy#L63). 

I think the bigger problem is that the scope of `build-plugin` has perhaps gotten a bit large. It really should be chopped up into several more focused plugins, for which a general "build plugin" then applies. This would allow us to ditch stuff [like this](https://github.com/elastic/elasticsearch/blob/2d9c898696d68e2a6e5c70f35cd304728a680276/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/StandaloneTestPlugin.groovy#L44). And subsequently apply only the subset of plugins we actually _need_ to projects like `build-tools`.